Base/Python/slicer/util.py

```diff
@@ -2334,90 +2422,13 @@ def pip_install(req):
-  except:
-    # Sequences module is not installed
-    pass
-
-def setMenuBarsVisible(visible, ignore=None):
-  """Show/hide all menu bars, except those listed in
-  ignore list."""
-  for menubar in mainWindow().findChildren('QMenuBar'):
-    if ignore is not None and menubar in ignore:
-      continue
-    menubar.setVisible(visible)
-
-def setPythonConsoleVisible(visible):
-  """Show/hide Python console."""
-  mainWindow().pythonConsole().parent().setVisible(visible)
-
-def setStatusBarVisible(visible):
-  """Show/hide status bar"""
-  mainWindow(verbose=False).statusBar().setVisible(visible)
-
-def setViewControllersVisible(visible):
-  """Show/hide view controller toolbar at the top of slice and 3D views"""
-  import slicer
-  lm = slicer.app.layoutManager()
-  for viewIndex in range(lm.threeDViewCount):
-    lm.threeDWidget(viewIndex).threeDController().setVisible(visible)
-  for sliceViewName in lm.sliceViewNames():
-    lm.sliceWidget(sliceViewName).sliceController().setVisible(visible)
-  for viewIndex in range(lm.tableViewCount):
-    lm.tableWidget(viewIndex).tableController().setVisible(visible)
-  for viewIndex in range(lm.plotViewCount):
-    lm.plotWidget(viewIndex).plotController().setVisible(visible)
-
-def forceRenderAllViews():
-  """Force rendering of all views"""
-  import slicer
-  lm = slicer.app.layoutManager()
-  for viewIndex in range(lm.threeDViewCount):
-    lm.threeDWidget(viewIndex).threeDView().forceRender()
-  for sliceViewName in lm.sliceViewNames():
-    lm.sliceWidget(sliceViewName).sliceView().forceRender()
-  for viewIndex in range(lm.tableViewCount):
-    lm.tableWidget(viewIndex).tableView().repaint()
-  for viewIndex in range(lm.plotViewCount):
-    lm.plotWidget(viewIndex).plotView().repaint()
+  :param string path: Path to be made safe if too long
 
-def setModulePanelTitleVisible(visible):
-  """Show/hide module panel title bar at the top of module panel.
-  If the title bar is not visible then it is not possible to drag and dock the
-  module panel to a different location."""
-  modulePanelDockWidget = mainWindow().findChildren('QDockWidget','PanelDockWidget')[0]
-  if visible:
-    modulePanelDockWidget.setTitleBarWidget(None)
-  else:
-    import qt
-    modulePanelDockWidget.setTitleBarWidget(qt.QWidget(modulePanelDockWidget))
-
-def setApplicationLogoVisible(visible):
-  """Show/hide application logo at the top of module panel."""
-  widget = findChild(mainWindow(), "LogoLabel")
-  widget.setVisible(visible)
-
-def setModuleHelpSectionVisible(visible):
-  """Show/hide Help section at the top of module panel."""
-  modulePanel = findChild(mainWindow(), "ModulePanel")
-  modulePanel.helpAndAcknowledgmentVisible = visible
-
-def setDataProbeVisible(visible):
-  """Show/hide Data probe at the bottom of module panel."""
-  widget = findChild(mainWindow(), "DataProbeCollapsibleWidget")
-  widget.setVisible(visible)
+  :return string: Safe path
+  """
+  return u"\\\\?\\" + path.replace('/', '\\')
```

Comment:
Didn't you want a platform check here?  Other places we use `if os.name == 'nt':`

Also just for reference PATH_MAX on macos is 1024 and NAME_MAX is 255 (in
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/syslimits.h)

Reply:
Yes, it will be needed. Since the application I use this in is Windows only, I submitted this as is for now, until we all agree that this will be necessary. @lassoan or @jcfr may have a different idea.